src/transcoder/transcoder.cpp

```diff
@@ -485,7 +484,8 @@ bool Transcoder::StartJob(const Job& job) {
     gst_element_link_many(convert, resample, muxer, sink, nullptr);
 
   // Set properties
-  g_object_set(src, "location", job.input.toUtf8().constData(), nullptr);
+  g_object_set(decode, "uri", job.input.toString().toUtf8().constData(),
+               nullptr);
```

Comment:
I have tested this now with files containing several special characters in the path: Encoding/decoding of those does not seem to be causing any problems for the transcoder.
I have used the opportunity to make the transcoder API now more consistent in its usage of URLs instead of filename. I've also changed the outputs from the transcoder dialog to display filepaths instead of the internally used URLs whenever it notices that the URL represents a local file so as not to confuse the user / be consistent with behavior until now.